include/GIL/Instructions/StructDestructureInst.hpp

```diff
@@ -0,0 +1,68 @@
+#ifndef GLU_GIL_INSTRUCTIONS_STRUCT_DESTRUCTURE_INST_HPP
+#define GLU_GIL_INSTRUCTIONS_STRUCT_DESTRUCTURE_INST_HPP
+
+#include "AggregateInst.hpp"
+#include "Member.hpp"
+#include "Type.hpp"
+
+namespace glu::gil {
+
+/// @class StructDestructureInst
+/// @brief Instruction that destructures a struct value into its individual
+/// fields.
+///
+/// It produces N results, where N is the number of fields in the struct.
+/// The only operand is the struct value.
+class StructDestructureInst : public AggregateInst {
+    Value _structValue; ///< The value of the struct being destructured.
+    std::vector<Member> _members; ///< The members (fields) of the struct.
+
+public:
+    /// @brief Constructs a StructDestructureInst.
+    ///
+    /// @param structValue The struct value to destructure.
+    /// @param members The list of struct members (fields).
+    StructDestructureInst(Value structValue, std::vector<Member> members)
```

Comment:
Members isn’t an operand, it shouldn’t be passed here, you can get it from structValue->getType->getType, llvm::cast to StructTy, then getFields